Missing strict dependency @angular/build -> @angular/core

### Command

test

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

This import statement 
https://github.com/angular/angular-cli/blob/a175d285d611678dd1f6d29de10c0a0dea09f780/packages/angular/build/src/builders/karma/polyfills/init_test_bed.js#L9
is not accompanied by a declaration of the dependency in 
https://github.com/angular/angular-cli/blob/main/packages/angular/build/package.json

Using `pnpm` with `hoist=false` in the `.npmrc` file, the `node_modules` tree is layed out in a strict way. When I build with `esbuild` it causes an error:

```
✘ [ERROR] Could not resolve "@angular/core/testing"

    ../../../../../../../node_modules/.aspect_rules_js/@angular+build@20.0.0-next.3_1309402193/node_modules/@angular/build/src/builders/karma/polyfills/init_test_bed.js:9:27:
      9 │ import { getTestBed } from '@angular/core/testing';
        ╵                            ~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@angular/core/testing" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.
```

The problem is observed in https://github.com/devversion/rules_angular/pull/19

### Minimal Reproduction

See this PR, it uses Bazel to run it. https://github.com/aspect-build/bazel-examples/pull/417

### Exception or Error

```text

```

### Your Environment

```text
"@angular/cli": "^20.0.0-next.3"
```

### Anything else relevant?

_No response_

Note, it's observable outside Bazel as well:

```
angular% ./node_modules/.bin/ng test my-lib
Application bundle generation failed. [2.290 seconds]

✘ [ERROR] Could not resolve "@angular/core/testing"

    ../node_modules/.pnpm/@angular+build@20.0.0-next.3_@angular+compiler-cli@20.0.0-next.4_@angular+compiler@20.0.0-nex_6bssrvcjsmwkxeaagiwva3myci/node_modules/@angular/build/src/builders/karma/polyfills/init_test_bed.js:9:27:
      9 │ import { getTestBed } from '@angular/core/testing';
        ╵                            ~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@angular/core/testing" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.


✘ [ERROR] Could not resolve "@angular/compiler"

    ../node_modules/.pnpm/@angular+core@20.0.0-next.4_rxjs@7.8.2_zone.js@0.15.0/node_modules/@angular/core/fesm2022/testing.mjs:11:31:
      11 │ import { ResourceLoader } from '@angular/compiler';
         ╵                                ~~~~~~~~~~~~~~~~~~~

  You can mark the path "@angular/compiler" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.

```

--- a/packages/angular/build/package.json
+++ b/packages/angular/build/package.json
@@ -57,6 +57,7 @@
     "rxjs": "7.8.2"
   },
   "peerDependencies": {
+    "@angular/core": "0.0.0-ANGULAR-FW-PEER-DEP",
     "@angular/compiler": "0.0.0-ANGULAR-FW-PEER-DEP",
     "@angular/compiler-cli": "0.0.0-ANGULAR-FW-PEER-DEP",
     "@angular/localize": "0.0.0-ANGULAR-FW-PEER-DEP",
@@ -69,9 +70,13 @@
     "ng-packagr": "0.0.0-NG-PACKAGR-PEER-DEP",
     "postcss": "^8.4.0",
     "tailwindcss": "^2.0.0 || ^3.0.0 || ^4.0.0",
+    "tslib": "^2.3.0",
     "typescript": ">=5.8 <5.9"
   },
   "peerDependenciesMeta": {
+    "@angular/core": {
+      "optional": true
+    },
     "@angular/localize": {
       "optional": true
     },

--- a/packages/angular_devkit/build_angular/package.json
+++ b/packages/angular_devkit/build_angular/package.json
@@ -73,6 +73,7 @@
     "undici": "7.7.0"
   },
   "peerDependencies": {
+    "@angular/core": "0.0.0-ANGULAR-FW-PEER-DEP",
     "@angular/compiler-cli": "0.0.0-ANGULAR-FW-PEER-DEP",
     "@angular/localize": "0.0.0-ANGULAR-FW-PEER-DEP",
     "@angular/platform-browser": "0.0.0-ANGULAR-FW-PEER-DEP",
@@ -90,6 +91,9 @@
     "typescript": ">=5.8 <5.9"
   },
   "peerDependenciesMeta": {
+    "@angular/core": {
+      "optional": true
+    },
     "@angular/localize": {
       "optional": true
     },

--- a/tests/legacy-cli/e2e/tests/misc/dedupe-duplicate-modules.ts
+++ b/tests/legacy-cli/e2e/tests/misc/dedupe-duplicate-modules.ts
@@ -11,7 +11,7 @@ export default async function () {
     await updateJsonFile('package.json', (json) => {
       json.dependencies = {
         ...json.dependencies,
-        'tslib': '2.0.0',
+        'tslib': '^2.0.0',
         'tslib-1': 'npm:tslib@1.13.0',
         'tslib-1-copy': 'npm:tslib@1.13.0',
       };
@@ -56,7 +56,7 @@ export default async function () {
       throw new Error('Expected stderr to contain [DedupeModuleResolvePlugin] log for tslib.');
     }
 
-    await expectFileToMatch(outFile, './node_modules/tslib/tslib.es6.js');
+    await expectFileToMatch(outFile, './node_modules/tslib/tslib.es6.mjs');
   } finally {
     await rimraf('node_modules/tslib');
     await gitClean();

